torchtext/csrc/vectors.cpp

```diff
@@ -11,101 +11,79 @@
 #include <stdexcept>
```

Comment:
Why not move those header files to `vectors.h` together

Reply:
I think in general, it is good practice to not include gratuitious `#include` directives. A header file should not be included in a `.h` file if the `.h` file itself doesn't need those other header files.

@cpuhrsch can correct me if I am wrong

Reply:
OK, thanks for the explanation.

Reply:
> I think in general, it is good practice to not include gratuitious `#include` directives. A header file should not be included in a `.h` file if the `.h` file itself doesn't need those other header files.
> 
> @cpuhrsch can correct me if I am wrong

I second that.